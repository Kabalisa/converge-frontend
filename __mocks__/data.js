const data = {
  office: {
    name: 'EPIC Tower',
    location: 'Nigeria',
    timezone: 'GMT +1',
  },
  amenity: {
    name: 'Marker',
    rooms: 4,
  },
};

module.exports = data;
