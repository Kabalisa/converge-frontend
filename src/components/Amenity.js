import React from 'react';
import PropTypes from 'prop-types';

const Amenity = ({ amenity: { name, rooms } }) => (
  <tr>
    <td>{name}</td>
    <td>{rooms}</td>
    <td>Edit &emsp; Delete</td>
  </tr>
);

Amenity.propTypes = {
  amenity: PropTypes.shape({
    name: PropTypes.string.isRequired,
    rooms: PropTypes.number.isRequired,
  }).isRequired,
};

export default Amenity;
