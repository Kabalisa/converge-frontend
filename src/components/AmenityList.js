import React from 'react';
import Amenity from './Amenity';
import DeleteResource from './DeleteResource';
import AddResource from './AddResource';
import '../assets/styles/amenitylist.scss';
import amenitiesList from '../fixtures/amenities';

class AmenityList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      amenities: amenitiesList,
      deleteModal: false,
    };
  }

  deleteResource = (amenity) => {
    this.setState({
      deleteModal: true,
      toDelete: amenity,
    });
  };

  handleCloseModal = () => {
    this.setState({ deleteModal: false });
  }

  render() {
    const { deleteModal, toDelete, amenities } = this.state;
    return (
      <div className="settings-amenity">
        {deleteModal && (<DeleteResource
          openModal={deleteModal}
          toDelete={toDelete}
          handleCloseModal={this.handleCloseModal}
        />)}
        <AddResource />
        <div className="settings-amenity-list">
          <table>
            <colgroup>
              <col className="first-col" />
              <col />
              <col className="last-col" />
            </colgroup>
            <thead>
              <tr>
                <th>Amenities</th>
                <th>Rooms</th>
                <th>Action</th>
              </tr>
            </thead>
            <tbody>
              {amenities.map(amenity => (
                <Amenity
                  amenity={amenity}
                  key={amenity.name}
                  doDelete={() => this.deleteResource(amenity)}
                />
              ))}
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

export default AmenityList;
