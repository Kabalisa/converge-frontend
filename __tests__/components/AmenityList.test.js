import React from 'react';
import { shallow, mount } from 'enzyme';

import AmenityList from '../../src/components/AmenityList';
import Amenity from '../../src/components/Amenity';
import data from '../../__mocks__/data';

describe('Tests for AmenityList Component', () => {
  const wrapper = shallow(<AmenityList />);
  const mountWrapper = mount(<AmenityList />);

  it('renders correctly from memory', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render `Add Amenities` button', () => {
    expect(mountWrapper.find('#modal-button')).toHaveLength(1);
    expect(mountWrapper.find('#modal-button').text()).toBe('Add Amenities');
  });
  it('should respond to delete click event', () => {
    wrapper.instance().deleteResource(data.amenity);
    expect(wrapper.state('deleteModal')).toEqual(true);
  });
  it('should close the modal', () => {
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('deleteModal')).toEqual(false);
  });
  it('should render the `Amenity`, which lists out amenities', () => {
    expect(mountWrapper.find(Amenity)).toHaveLength(6);
  });
});
