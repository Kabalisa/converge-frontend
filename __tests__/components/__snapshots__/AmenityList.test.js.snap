// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AmenityList Component renders correctly from memory 1`] = `
<div
  className="settings-amenity"
>
  <AddResource />
  <div
    className="settings-amenity-list"
  >
    <table>
      <colgroup>
        <col
          className="first-col"
        />
        <col />
        <col
          className="last-col"
        />
      </colgroup>
      <thead>
        <tr>
          <th>
            Amenities
          </th>
          <th>
            Rooms
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <Amenity
          amenity={
            Object {
              "name": "Apple TV",
              "rooms": 18,
            }
          }
          doDelete={[Function]}
          key="Apple TV"
        />
        <Amenity
          amenity={
            Object {
              "name": "Jabra Speaker",
              "rooms": 26,
            }
          }
          doDelete={[Function]}
          key="Jabra Speaker"
        />
        <Amenity
          amenity={
            Object {
              "name": "White Board",
              "rooms": 18,
            }
          }
          doDelete={[Function]}
          key="White Board"
        />
        <Amenity
          amenity={
            Object {
              "name": "Marker",
              "rooms": 24,
            }
          }
          doDelete={[Function]}
          key="Marker"
        />
        <Amenity
          amenity={
            Object {
              "name": "Air Conditioner",
              "rooms": 14,
            }
          }
          doDelete={[Function]}
          key="Air Conditioner"
        />
        <Amenity
          amenity={
            Object {
              "name": "Chair",
              "rooms": 18,
            }
          }
          doDelete={[Function]}
          key="Chair"
        />
      </tbody>
    </table>
  </div>
</div>
`;
